### PR TITLE
IW-3045 | Forum migration fixes

### PR DESCRIPTION
**Pull request by pstacherzak**
@Wikia/iwing

### CODE REVIEW

File: extensions/wikia/Discussions/maintenance/ForumDumper.php [hunk 1]
```diff
@@ -771,7 +788,7 @@ private function getRelatedArticleData( $textId ) {
 			return [null, null];
 		}
 
-		return [ $title->getText(), $title->getLocalURL() ];
```
Comment (rybmat) [inline extensions/wikia/Discussions/maintenance/ForumDumper.php:774]: why this change?

Comment (pstacherzak) [reply extensions/wikia/Discussions/maintenance/ForumDumper.php:774]: The initial problem was that some urls were too long for `relative_url` column (which is 261 varchar - 255 for title and 6 for /wiki/ prefix) for article_tags. `$title->getLocalURL()` gives me correct url, but in percentage escaping form, so for some cases url become longer than 261 characters (like changing one character `'` to `%27` 3 characters). I have checked how it works manually outside of migration context and it looks like url in `relative_url` column is just a wiki prefix + title (with spaces and all special characters).

Comment (rybmat) [reply extensions/wikia/Discussions/maintenance/ForumDumper.php:774]: but what about language path part?

Comment (rybmat) [reply extensions/wikia/Discussions/maintenance/ForumDumper.php:774]: how about some decoding function over getLocalUrl?

Comment (pstacherzak) [reply extensions/wikia/Discussions/maintenance/ForumDumper.php:774]: It is not presented in `relative_url` table, so it must be evaluated somehow later. I have checked it on https://nikodev.fandom.com/de/f article tags

Comment (pstacherzak) [reply extensions/wikia/Discussions/maintenance/ForumDumper.php:774]: About `urldecode` - I am not sure if it is a correct way, because `getLocalURL` returns a link with spaces replaced with dash as far as I noticed. I did a check for spaces in database and it looks like there are spaces in `relative_url` column in article_tags, just like in the title.

**Review by rybmat — APPROVED** [review]

**Review by rybmat — APPROVED** [review]
